File: app-python/entitlements_app/services/list_account_overrides_query_service.py
"""List account-specific entitlement overrides."""

from decimal import Decimal

from entitlements_app.requests.list_account_overrides_query_request import ListAccountOverridesQueryRequest
from entitlements_app.entities import AccountOverride
from entitlements_app.ports import EntitlementsRepository


class ListAccountOverridesQueryService:
    """Service handling Query `ListAccountOverrides`."""

    def __init__(self, entitlements_repository: EntitlementsRepository) -> None:
        self.entitlements_repository: EntitlementsRepository = entitlements_repository

    def handle(self, request: ListAccountOverridesQueryRequest) -> list[AccountOverride]:
        # spec-driven body for Query ListAccountOverrides.
        # intent: List account-specific entitlement overrides.
        # reads: EntitlementsRepository.listOverridesByAccount via `overrides`.
        return self.entitlements_repository.listOverridesByAccount(request.accountId)

File: app/src/main/java/com/example/spring/ListAccountOverridesQueryService.java
package com.example.spring;

import com.example.spring.error.NotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * List account-specific entitlement overrides.
 */
@Service
public class ListAccountOverridesQueryService {

    private final EntitlementsRepository entitlementsRepository;

    public ListAccountOverridesQueryService(EntitlementsRepository entitlementsRepository) {
        this.entitlementsRepository = entitlementsRepository;
    }

    @Transactional(readOnly = true)
    public java.util.List<AccountOverride> handle(ListAccountOverridesQueryRequest request) {
        // === GEN_REGION_START: service-body ===
        // spec-driven body for Query ListAccountOverrides.
        // intent: List account-specific entitlement overrides.
        // reads: EntitlementsRepository.listOverridesByAccount via `overrides`.
        return this.entitlementsRepository.listOverridesByAccount(request.accountId());
        // === GEN_REGION_END: service-body ===
    }
}
